CspSubmissionPlugin.inc.php

```diff
@@ -1166,8 +1166,24 @@ public function submissionfilesuploadform_display($hookName, $args)
 			$templateMgr->setData('submissionFileGenres', $genreList);
 
 		}
-		if ($fileStage == 10) { // UPLOAD DE PDF PARA PUBLICAÇÃO
-			$templateMgr->setData('isReviewAttachment', TRUE); // SETA A VARIÁVEL PARA TRUE POIS ELA É VERIFICADA NO TEMPLATE PARA NÃO EXIBIR OS COMPONENTES
+		if ($fileStage == 10) { // Upload de PDF para publicação
+			$result = $userDao->retrieve(
+				<<<QUERY
+				SELECT A.genre_id, setting_value
+				FROM ojs.genre_settings A
+				LEFT JOIN ojs.genres B
+				ON B.genre_id = A.genre_id
+				WHERE locale = '$locale' AND entry_key LIKE 'EDITORACAO_DIAGRM_PDF_PUBL%'
+				QUERY
+			);
+
+			while (!$result->EOF) {
+				$genreList[$result->GetRowAssoc(0)['genre_id']] = $result->GetRowAssoc(0)['setting_value'];
```

Comment:
essa forma de lidar com o retorno da query, é uma coisa do OJS? não da pra fazer um foreach e usar o array direto, tipo o que eu fiz (aqui)[https://github.com/LyseonTech/ojs-csp-submission/pull/12/files#diff-fe31a2e2a2c80e5149015e6f089c83d0f16be0cece20e2d95738e43bdad63233R87]?

Reply:
pra performance talvez pegue um pouco pois tá chamando 2x o método GetRowAssoc, não sei o que ele faz mas dependendo do que for pode onderar a aplicação.